src/connector/src/sink/deltalake.rs

```diff
@@ -0,0 +1,544 @@
+            return Err(SinkError::DeltaLake(format!(
+                "column count not match, rw is {}, deltalake is {}",
+                self.param.schema().fields().len(),
+                deltalake_fields.len()
+            )));
+        }
+        for field in self.param.schema().fields() {
+            if !deltalake_fields.contains_key(&field.name) {
+                return Err(SinkError::DeltaLake(format!(
+                    "column {} not found in deltalake table",
+                    field.name
+                )));
+            }
+            let deltalake_field_type: &&DeltaLakeDataType = deltalake_fields
+                .get(&field.name)
+                .ok_or_else(|| SinkError::DeltaLake("cannot find field type".to_string()))?;
+            if !check_field_type(&field.data_type, deltalake_field_type)? {
+                return Err(SinkError::DeltaLake(format!(
+                    "column {} type is not match, deltalake is {:?}, rw is{:?}",
+                    field.name, deltalake_field_type, field.data_type
+                )));
+            }
+        }
+        Ok(())
+    }
+
+    async fn new_coordinator(&self) -> Result<Self::Coordinator> {
+        Ok(DeltaLakeSinkCommitter {
+            table: self.config.common.create_deltalake_client().await?,
+        })
+    }
+}
+
+impl TryFrom<SinkParam> for DeltaLakeSink {
+    type Error = SinkError;
+
+    fn try_from(param: SinkParam) -> std::result::Result<Self, Self::Error> {
+        let config = DeltaLakeConfig::from_hashmap(param.properties.clone())?;
+        DeltaLakeSink::new(config, param)
+    }
+}
+
+pub struct DeltaLakeSinkWriter {
+    pub config: DeltaLakeConfig,
+    schema: Schema,
+    pk_indices: Vec<usize>,
+    writer: RecordBatchWriter,
+    is_append_only: bool,
+    dl_schema: Arc<deltalake::arrow::datatypes::Schema>,
+    dl_table: DeltaTable,
+}
+
+impl DeltaLakeSinkWriter {
+    pub async fn new(
+        config: DeltaLakeConfig,
+        schema: Schema,
+        pk_indices: Vec<usize>,
+        is_append_only: bool,
+    ) -> Result<Self> {
+        let dl_table = config.common.create_deltalake_client().await?;
+        let writer = RecordBatchWriter::for_table(&dl_table)?;
+        let dl_schema: Arc<deltalake::arrow::datatypes::Schema> =
+            Arc::new(covert_schema(dl_table.get_schema()?)?);
+
+        Ok(Self {
+            config,
+            schema,
+            pk_indices,
+            writer,
+            is_append_only,
+            dl_schema,
+            dl_table,
+        })
+    }
+
+    async fn write(&mut self, chunk: StreamChunk) -> Result<()> {
+        let a = to_record_batch_with_schema(self.dl_schema.clone(), &chunk)
+            .map_err(|err| SinkError::DeltaLake(format!("convert record batch error: {}", err)))?;
+        self.writer.write(a).await?;
+        Ok(())
+    }
+}
+
+fn covert_schema(schema: &StructType) -> Result<deltalake::arrow::datatypes::Schema> {
```

Comment:
topo. `covert` -> `convert`.

This method looks like a general utils method. Is this method implemented in the delta lake sdk?

Reply:
We only found the conversion method for datatype, so we need to write a loop iteration ourselves.

Reply:
I just found that we can use `deltalake::arrow::datatypes::Schema::try_from(dl_table.get_schema()?)` in where we currently call `convert_schema`.